Fix test_add_old_resource API test

### Description 

The `test_add_old_resource` seems to fail due to this does not autoconfigure, making it dependent of the state of the API in the previous tests

https://ci.wazuh.info/view/Tests/job/Test_integration/17631/console

```
  test_api/test_rbac/conftest.py:20: in set_security_resources
      api_details = get_api_details_dict()
  /usr/local/lib/python3.6/site-packages/wazuh_testing-4.3.0-py3.6.egg/wazuh_testing/api.py:71: in get_api_details_dict
      sleep_time)
  /usr/local/lib/python3.6/site-packages/wazuh_testing-4.3.0-py3.6.egg/wazuh_testing/api.py:65: in get_token_login_api
      raise RuntimeError(f"Error obtaining login token: {response.json()}")
  E   RuntimeError: Error obtaining login token: {'title': 'Wazuh Internal Error', 'detail': 'Timeout executing API request', 'dapi_errors': {'node01': {'error': 'Timeout executing API request', 'logfile': 'WAZUH_HOME/logs/api.log'}}, 'error': 3021}
```

--- a/tests/integration/test_api/conftest.py
+++ b/tests/integration/test_api/conftest.py
@@ -99,3 +99,18 @@ def wait_for_start(get_configuration, request):
 @pytest.fixture(scope='module')
 def get_api_details():
     return get_api_details_dict
+
+
+@pytest.fixture(scope='module')
+def restart_api_module(request):
+    # Stop Wazuh and Wazuh API
+    control_service('stop')
+    control_service('start')
+
+
+@pytest.fixture(scope='module')
+def wait_for_start_module(request):
+    # Wait for API to start
+    file_monitor = FileMonitor(API_LOG_FILE_PATH)
+    file_monitor.start(timeout=20, callback=callback_detect_api_start,
+                       error_message='Did not receive expected "INFO: Listening on ..." event')

--- a/tests/integration/test_api/test_rbac/test_add_old_resource.py
+++ b/tests/integration/test_api/test_rbac/test_add_old_resource.py
@@ -66,7 +66,7 @@ user_id, role_id, policy_id, rule_id = None, None, None, None
 
 # Tests
 @pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
-def test_add_old_user(set_security_resources, get_api_details):
+def test_add_old_user(restart_api_module, wait_for_start_module, set_security_resources, get_api_details):
     '''
     description: Check if the security relationships of a previous user are maintained
                  in the system after adding a new user with the same ID.

